matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/pushrules/EventMatchCondition.kt

```diff
@@ -31,40 +29,36 @@ class EventMatchCondition(
          * The glob-style pattern to match against. Patterns with no special glob characters should
          * be treated as having asterisks prepended and appended when testing the condition.
          */
-        val pattern: String,
-        /**
-         * true to match only words. In this case pattern will not be considered as a glob
-         */
-        val wordsOnly: Boolean
+        val pattern: String
 ) : Condition {
 
     override fun isSatisfied(event: Event, conditionResolver: ConditionResolver): Boolean {
         return conditionResolver.resolveEventMatchCondition(event, this)
     }
 
-    override fun technicalDescription() = "'$key' matches '$pattern', words only '$wordsOnly'"
+    override fun technicalDescription() = "'$key' matches '$pattern'"
 
     fun isSatisfied(event: Event): Boolean {
         // TODO encrypted events?
         val rawJson = MoshiProvider.providesMoshi().adapter(Event::class.java).toJsonValue(event) as? Map<*, *>
                 ?: return false
         val value = extractField(rawJson, key) ?: return false
 
-        // Patterns with no special glob characters should be treated as having asterisks prepended
-        // and appended when testing the condition.
+        // The match is performed case-insensitively, and must match the entire value of
+        // the event field given by `key` (though see below regarding `content.body`). The
+        // exact meaning of "case insensitive" is defined by the implementation of the
+        // homeserver.
+        //
+        // As a special case, if `key` is `content.body`, then `pattern` must instead
+        // match any substring of the value of the property which starts and ends at a
+        // word boundary.
         return try {
-            if (wordsOnly) {
-                value.caseInsensitiveFind(pattern)
-            } else {
-                val modPattern = if (pattern.hasSpecialGlobChar()) {
-                    // Regex.containsMatchIn() is way faster without leading and trailing
-                    // stars, that don't make any difference for the evaluation result
-                    pattern.removePrefix("*").removeSuffix("*").simpleGlobToRegExp()
-                } else {
-                    pattern.simpleGlobToRegExp()
-                }
-                val regex = Regex(modPattern, RegexOption.DOT_MATCHES_ALL)
+            if (key == "content.body") {
+                val regex = Regex("(\\W|^)" + pattern.simpleGlobToRegExp() + "(\\W|$)", setOf(RegexOption.DOT_MATCHES_ALL, RegexOption.IGNORE_CASE))
```

Comment:
How different from caseInsensitiveFind is this? can it reuse it too?

Reply:
caseInsesitiveFind doesn't evaluate globs (since it escapes the string, from what I can tell), otherwise it is the same.